docs/program/apis/sessions.md

```diff
@@ -0,0 +1,88 @@
+---
+title: "Sessions Management with Viam's Client SDKs"
+linkTitle: "Session Management API"
+weight: 20
+type: "docs"
+description: "How to use the Sessions Management API with Viam's Client SDKs."
+tags: ["client", "sdk", "viam-server", "networking", "apis", "robot api", "session", "sessions", "session management"]
+---
+
+The Session Management API provides support for robot session management.
+
+A Session allows a **client** to express that it is actively connected or authenticated to `viam-server` on your robot and
+supports stopping actuating components when the connection is no longer active.
+
+{{< alert title="Info" color="info" >}}
+
+### Purpose of Session Management
+
+When controlling a robot or fleet with the Viam platform, you want some kind of protocol or system-wide means to understand the presence of the clients that are talking and authenticated to each robot's `viam-server` through Viam's various [resource APIs](/program/apis/).
+
+The presence of these clients at one point in time can be referred to as a "session".
+A session technically is any presence mechanism at the RDK application layer maintained by any client with `viam-server`.
+Session management, as explained here, allows for safer robotic operation when you are dealing with moving parts on the robot (the "actuating controls").
+
+Specifically, without session management, controls that would be "sticky" (for example [`SetPower()` of a base](/components/base/#setpower)) based on the last input of a **client**, meaning the API request from the client sets the flow of electricity on a part of the robot and then does not time out, could lead to **having a robot trying to continue doing what it was told to do forever.**
+
+You do not want that to happen.
+
+#### Clients and Viam's Client SDKs
+
+A client has multiple meanings for a Viam robot.
+Essentially, it's anything that is receiving the information served by `viam-server` running on the robot.
+
+A **client** can be an SDK script controlling the robot, an input controller, or just the different resources on the robot talking amongst themselves.
+For example, if you use Viam's module registry to [add modular resources to your robot](/extend/modular-resources/), the clients of your robot in its lifetime will include the "model servers" you instantiate on your robot for individual resources.
+
+Viam's Session Management API is your built-in solution to this.
+Your client maintains the session, telling the `viam-server` instance that it is still present every so often, or staying within the heartbeat window.
+
+Your client must send at least one session heartbeat within this window.
```

Comment:
need to add more info about how to send a session heartbeat likely if this stays here on line 40

Reply:
Okay; I think there's a way to transition from the part about what happens without session management into this part nicely, but will wait to make suggestions until after this is done.

Reply:
Heartbeats are sent automatically from our clients (golang, python, and typescript) unless disabled via a client option or session management is not implemented by the server. 

As of now, heartbeats are sent at an interval that is 1/5th of the heartbeat window (e.g. if the heartbeat window is 5 seconds, clients will send a heartbeat every 1 second).

Reply:
@JessamyT What would you think about me making this part a separate page/a second-half ticket of this? wondering if could merge as is to get this info up and then add a lot more about heartbeats, but maybe i should just do it now lol

Reply:
I don't think it needs to be a whole second page? Maybe just incorporate it now? I think some of this stuff can be kinda combined/made smooth and succinct, and the basics of heartbeats are sorta part of how the whole thing works so it's important to at least have high level info about that.

Reply:
But if you think there need to be lots of details added, sure, can be a separate PR to modify this page.

Reply:
yeah i just want to get this part up first tbh but i can find time to add this in 🤷‍♀️ That makes sense that it's fundamental

Reply:
Will comment here when done

Reply:
Ok @maximpertsov @JessamyT I think I have added the necessary info for users for client heartbeats now, sorry for delay here